style/layer/ferry.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+// Filter properties in this layer should be updated to reflect consensus once
+// https://github.com/openmaptiles/openmaptiles/issues/1373 is closed
+
+export const ferry = {
+  id: "ferry",
+  type: "line",
+  paint: {
+    "line-color": "hsl(211, 30%, 38%)",
+    "line-dasharray": [7, 5],
+    "line-width": 1.5,
+  },
+  filter: ["any", ["==", "class", "ferry"], ["==", "subclass", "ferry"]],
+  layout: {
+    visibility: "visible",
+  },
+  source: "openmaptiles",
+  "source-layer": "transportation",
+};
+
+export const ferryLabel = {
+  id: "ferry_label",
+  type: "symbol",
+  filter: ["any", ["==", "class", "ferry"], ["==", "subclass", "ferry"]],
+  paint: {
+    "text-color": "hsl(211, 53%, 15%)",
+    "text-halo-color": "hsl(211, 70%, 90%)",
+    "text-halo-blur": 0.5,
+    "text-halo-width": 1.5,
```

Comment:
If we can be pretty sure these labels will always be over water, do they need halos? The dark blue text probably has enough contrast against the light blue water, just as the ferry lines themselves do.

Reply:
I initially thought the same thing, but without the halo the text becomes hard to read in places where there are several dense ferry routes and some text may be rendered on top of other route lines

Reply:
Example with no halo looking janky on top of nearby lines
![image](https://user-images.githubusercontent.com/281482/162102654-50929d33-085a-4ecd-948a-b9c168fac12d.png)

Setting halo color the same as water fill color produces a knockout effect that looks better.  Looks kind of odd to have a halo on the water labels but not to have one on the ferry labels though.
![image](https://user-images.githubusercontent.com/281482/162103011-622ad81b-d20f-4c08-b403-e1aa43e028b6.png)

Knockouts for both label types could be an option.  Gives a bit of a different look though.
![image](https://user-images.githubusercontent.com/281482/162103504-d00e22d0-aade-48d1-b156-d765d10d05c7.png)

Reply:
Except the faux knockout effect breaks down when a water label ends up partially over land.  So that's not really an option.
![image](https://user-images.githubusercontent.com/281482/162104297-dd03eb17-25bf-4e20-91b5-e9590c5d674b.png)

Reply:
> Setting halo color the same as water fill color produces a knockout effect that looks better. Looks kind of odd to have a halo on the water labels but not to have one on the ferry labels though.

This knockout effect is much less distracting; thanks for trying it out. I agree that it’s inconsistent to keep the halo on the water body labels, but those labels very often overlap non–water bodies, as you’ve seen. Also, I think the water body would tend to be more notable than the ferry routes that traverse it, so the added prominence is justifiable.

Reply:
I guess halos on the water labels and knockouts on the ferry labels isn't as weird as I initally thought and your right that they are less distracting.  I think the halos look good too, though.  Can we be sure these labels will rarely be over land?  I've found a couple of cases where they do go over a bit of land.  For example:
![image](https://user-images.githubusercontent.com/281482/162108208-7619d048-c66a-4be6-aa16-bcd0eb1e5b12.png)
It's not terribly noticeable, but might be more of an issue if a ferry route goes close to the shore for longer.  The halos avoid this.

Reply:
What if we have these knockouts fade like the halos? Then it wouldn’t look like the label took a bite out of the peninsula, or at least it would be less decisive in that regard.

Reply:
It's tricky.  Maplibre doesn't seem to leave much space around the letters for the halo to fade.  It quickly starts looking blocky at about 2px.  For the faux knockout I had the halo width set to 3px which looks basically like a rectangle if over a different color but over the same color you don't see it.  reducing the width and increasing the blur makes it look better over a different color, but makes the faux knockout much less effective at de-cluttering the lines behind the text. 🤷‍♂️

Reply:
Thanks for the renders @zekefarwell! I think it's fair to bring up accessibility considerations here too: dark blue on lighter blue (knockout) is less contrasting than dark blue on white-gray (halo)

Reply:
Halos can also impair legibility as much as a lack of contrast, especially when combined with tight letter spacing and italics. The halos pictured above are more legible than what we fixed in #203; however, if contrast is an issue, then it might be more effective to make the text black, as seen on some paper maps that nonetheless have blue ferry lines.

Reply:
> I guess we should be ok with the opposite (label taking a bite out of land) as well

This approach seems sensible to me!

Reply:
Clear as mud! 

In general I would avoid translucency as layering can be pretty unexpected.  Are we still playing with the knockout halo width?

Reply:
I'm happy with it as-is, but if other folks want it tweaked a bit feel free to propose new values or maintainers can add a commit 😃